Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -226,6 +248,94 @@ HexEditorWidget::HexEditorWidget()
     m_editor->set_focus(true);
 }
 
+void HexEditorWidget::update_inspector_values(size_t position)
+{
+    // build out primitive types like u8, i8, u16, etc
+    size_t byte_read_count = 0;
+    u64 unsigned_64_bit_int = 0;
+    for (int i = 0; i < 8; ++i) {
+        Optional<u8> read_result = m_editor->get_byte(position + i);
+        u8 current_byte = 0;
+        if (!read_result.has_value())
+            break;
+
+        current_byte = read_result.release_value();
+        if (m_value_inspector_little_endian)
+            unsigned_64_bit_int = ((u64)current_byte << (8 * byte_read_count)) + unsigned_64_bit_int;
+        else
+            unsigned_64_bit_int = (unsigned_64_bit_int << 8) + current_byte;
+
+        ++byte_read_count;
+    }
+
+    if (!m_value_inspector_little_endian) {
+        // if we didn't read far enough, lets finish shifting the bytes so the code below works
+        size_t bytes_left_to_read = 8 - byte_read_count;
+        unsigned_64_bit_int = (unsigned_64_bit_int << (8 * bytes_left_to_read));
+    }
+
+    // Populate the model
+    ValueInspectorModel* m_value_inspector_model = new ValueInspectorModel();
+    if (byte_read_count >= 1) {
+        u8 unsigned_byte_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_byte_value = (unsigned_64_bit_int & 0xFF);
+        else
+            unsigned_byte_value = (unsigned_64_bit_int >> (64 - 8)) & 0xFF;
+
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedByte, String::number((i8)unsigned_byte_value));
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedByte, String::number(unsigned_byte_value));
+    } else {
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedByte, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedByte, "");
+    }
+
+    if (byte_read_count >= 2) {
+        u16 unsigned_short_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_short_value = (unsigned_64_bit_int & 0xFFFF);
+        else
+            unsigned_short_value = (unsigned_64_bit_int >> (64 - 16)) & 0xFFFF;
+
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedShort, String::number((i16)unsigned_short_value));
```

Comment:
(same thing here)

Reply:
variable rename applied and switched to static_cast<>()